### ISSUE TITLE
Rethink interface

### ISSUE THREAD

**Issue by tdicola** [adafruit/Adafruit_CircuitPython_MPR121#1]
Let's look at remaking the interface to less directly follow the Arduino and Raspberry Pi Python library (which also should probably be updated in the same way and eventually share the same codebase).  Using properties to represent touch state might be more in line with python style.

Also consider for the touched function a variation that returns a 12-tuple of bools.  This would allow a quick single I2C call lookup of all the touch state and make it easy to parse out results, like:
```
touched_pins = mpr121.touched_pins()  # Returns 12-tuple of bools
if touched_pins[3]:
    print('Pin 4 touched!')
```

**Comment by tannewt** [comment]
@caternuson or @kattni this sounds up your allies.

**Comment by caternuson** [comment]
The MPR121 has pads labeled with numbers 0 through 11. I'm thinking there should be an individual property for each one with names `touch_0` through `touch_11`.

However, I'm seeing a couple of different syntaxes for touch being used for touch related properties.

**CPX**:
```python
if cpx.touch_A1:
    print('Touched pad A1')
```
**Crickit**:
```python
if crickit.touch_1.value:
    print("Touched Cap Touch Pad 1")
```
**touchio**
```python
if touch.value:
    print("Touched!")
```

What should be done here?

**Comment by tannewt** [comment]
I like the latter two for this library because then the channel object can be passed around.

cpx is different because its at the highest level of abstraction where you are assumed to be using it directly.

**Comment by ladyada** [comment]
@tannewt TERRIBLE THOUGHT, maybe intstead of having 12 properties, we could use the array indexing so it returns an introspective class that can return the value:
```if mpr121.touch[5].value: do_stuff```
or could
```
touchobject = mpr121.touch[5]
print(touchobject.raw_value)
```

**Comment by tannewt** [comment]
@ladyada Good idea! I think @caternuson did something similar for an ADC chip.

**Comment by ladyada** [comment]
@caternuson would that be ok by you ^? cause then we could maybe get closer to consistancy across all touch devices/interfaces :)

**Comment by caternuson** [comment]
I'm fine with either and can code up either. I am in favor of consistency though. I think that is critically important given target audience for CP.

Here's the ADC @tannewt mentioned:
https://github.com/adafruit/Adafruit_CircuitPython_ADS1x15
**BUT** there's a chance I was going to change the way that worked based on the more recent development being done for the MCP3xxxx series ADC by @brentru which took a different approach:
https://github.com/brentru/Adafruit_CircuitPython_MCP3xxx/blob/mcp3008-work/examples/mcp3xxx_simpletest.py#L19
@ladyada I think you and @brentru were working that one out? Another example where I'm fine with either approach, but just want them to be consistent.

Would you change Crikict to match? Otherwise we would be presenting two different looking syntaxes:
```python
crickit.touch_1.value
```
vs.
```python
mpr121.touch[5].value
```
A beginner would look at those two and wonder why one has square brackety things and the other has an underline thing.

**Comment by ladyada** [comment]
yeah we can go back and change crickit. everything is inconsistant right now anways so i dont care if it isnt a 'clean sweep' thru all the captouch libraries :)

**Comment by caternuson** [comment]
Cool. Than I guess it could just be a question of which syntax is more beginner friendly? I'm a fan of the syntax the array indexing provides.

**Comment by ladyada** [comment]
i like array indexing myself - there's rarely just one captouch

**Comment by caternuson** [comment]
OK. I'll go that route then.

@tannewt Any other comments?

**Comment by tannewt** [comment]
Nope. I agree!

**Comment by dhalbert** [comment]
I suggested indexing before for some of the crickit stuff but was outvoted at the time :smile:

**Comment by ladyada** [comment]
i think some crickit stuff we didnt have space for at the time, but we could try it now that we've optimized some stuff. i was really running out of room :(

**Comment by caternuson** [comment]
It does require creating a helper class that gets passed back by `__getitem__`. So it's potentially not as code compact as just having individual properties for each.

**Comment by caternuson** [comment]
Do we even need the `.touch` part? Couldn't it just be:
```python
mpr121[0].value
```

**Comment by ladyada** [comment]
thats fine, we probably want value and raw_value

**Comment by caternuson** [comment]
@ladyada for `raw_value` do you want "Electrode Filtered Data" from registers 0x04~0x1D?

**Comment by ladyada** [comment]
yep! this library will replace https://github.com/adafruit/Adafruit_Python_MPR121/blob/master/Adafruit_MPR121/MPR121.py and we use both filtered data and the 'touch status'! we'll also want to add a propertty setter/getter for 'threshold'

**Comment by caternuson** [comment]
Yep. Saw those thresholds - there are two, touch and release, for each channel! And there are some other knobs that would be nice to have access to. I think doing this all in one PR might be a bit much - the existing funcs could stand a re-write and some others will need to be added.

OK to have the initial PR just be adding `value` and `raw_value` access? Then we can create issues for adding the additional stuff.

**Comment by ladyada** [comment]
yup!

**Comment by caternuson** [comment]
Closing this one. Fixed by #9. Additional interface updates will be addressed in other issues / PRs.
